### PR TITLE
Adding milestone 03

### PR DESCRIPTION
**Pull request by Madhuka**
Feature List
- Make library and charts plug-gable (Remove hard coded code, and simulate loading libraries)
- Fixes and code improvement
- Bigger Test Cover
- Scenario Tests
- Remove unused files

Readme
- [x] Add all the steps: Bower install and npm install are missing
  UI:
- [x] No button style (just text) when we run grunt serve
- [x] Need similar width for all charts (NVD3 is smaller than others for example)
- [x] Need to have meaningful legends/Axis
  Bugs:
- [x]  After selecting a dataset, then a chart library, it will draw the chart. However if you click again on the same chart type that the one shown, there won't be any data in the chart.

Code Style:
- [x]  Fix trailing spaces
- [x]  Make code on multiple lines when html line are too long (~100-120 columns)
- [x] Fix google-chart-factory indentation
- [x] Fix Typos

Html reduction:
- [x] Reduce HTML lines repetitions, use ng-repeat with a configuration object instead
- [x] Remove unused files

Make code scalable for pluggable libraries:
- [x] Remove hard coded library names etc... from javascript code. Only an Array with Library Names should be used as configuration to refer to.
- [x] Include the Chart Html only when needed
  -> cf. ng-include, and https://github.com/apache/incubator-zeppelin/blob/master/zeppelin-web/app/index.html line 51 for example
  Fix a few bad things:
- [x] Remove the dummy Data from chart factories
- [x] loadData should not be inside of ChartCtrl
- [x] Change hardcoded loadChartLibrary function parameter inside the HTML: loadChartLibrary(2) -> should be loadChartLibrary($index) or (chart.name)
  Tests:
- [ ] Implement more tests, and change code if necessary
- [ ] Implement some scenario type of tests

### CODE REVIEW

File: app/scripts/chart-factory/google-chart-factory.js [hunk 1]
```diff
@@ -85,30 +66,22 @@ formatters : {}
   function getGoogleChart(error, rows) {
     GoogelChartChartModel.data.rows = rows;
   }
-
   var googlexChart = {
     model: googleChartModel,
     get: getGoogleChart
   };
 
-   function setChatTypeView(chartType){
```
Comment (corneadoug) [inline app/scripts/chart-factory/google-chart-factory.js:94]: Typo: setChatTypeView (x6)

File: app/scripts/chart-factory/nvd3-chart-factory.js [hunk 2]
```diff
@@ -44,21 +44,9 @@ angular.module('apacheZeppelinGsocApp').factory('NVD3ChartFactory', function(Cha
     data: [{
```
Comment (corneadoug) [inline app/scripts/chart-factory/nvd3-chart-factory.js]: still some dummy data?

File: app/scripts/chart-factory/google-chart-factory.js [hunk 3]
```diff
@@ -7,70 +6,52 @@
...
-    'isStacked': 'true',
-    'fill': 20,
-    'height': 300,
-    'displayExactValues': true,
-    'vAxis': {
-    'gridlines': {
-    'count': 6
-}
+    cssStyle: 'height:400px; width:500px;',
+    data: {
```
Comment (corneadoug) [inline app/scripts/chart-factory/google-chart-factory.js:20]: Still some dummy data

Comment (Madhuka) [inline app/scripts/chart-factory/google-chart-factory.js:20]: Those lines were there for model building.  When It is done It is shown in comment

Comment (corneadoug) [inline app/scripts/chart-factory/google-chart-factory.js:20]: So is it going to disappear?

Comment (Madhuka) [inline app/scripts/chart-factory/google-chart-factory.js:20]: data and series will removed but options like height, width will be there.

On Wed, Jun 24, 2015 at 11:04 AM, CORNEAU Damien notifications@github.com
wrote:

> In app/scripts/chart-factory/google-chart-factory.js
> https://github.com/Madhuka/apache-zeppelin-gsoc/pull/5#discussion_r33118408
> :
> 
> > -}, {
> > -    v: 10
> >   -}]
> > -    }]
> >   -},
> > -  options : {
> > -    'isStacked': 'true',
> > -    'fill': 20,
> > -    'height': 300,
> > -    'displayExactValues': true,
> > -    'vAxis': {
> > -    'gridlines': {
> > -    'count': 6
> >   -}
> > -    cssStyle: 'height:400px; width:500px;',
> > -    data: {
> 
> So is it going to disappear?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Madhuka/apache-zeppelin-gsoc/pull/5/files#r33118408.

## 

Cheers,
Madhuka Udantha
http://madhukaudantha.blogspot.com

**Comment by Madhuka** [comment]
Final UIs 
mz1

mz2

mz3

**Pull request merged**